### ISSUE TITLE
MNIST TensorFlow Website Example problem

### ISSUE THREAD

**Issue by adonisDias** [tensorflow/tensorflow#3693]
Hi! When i run the example of MNIST Deep model (got on the tensorflow website) in a jupyter notebook i get this error: The kernel appears to have died. It will restart automatically.
This happens in the line: 
print("test accuracy %g"%accuracy.eval(feed_dict={x: mnist.test.images, y_: mnist.test.labels, keep_prob: 1.0}))

COMPLETE CODE TENSORFLOW WEBSITE:
from tensorflow.examples.tutorials.mnist import input_data
mnist = input_data.read_data_sets("MNIST_data/", one_hot=True)

from __future__ import absolute_import
from __future__ import division
from __future__ import print_function

import gzip
import os
import tempfile

import numpy
from six.moves import urllib
from six.moves import xrange  # pylint: disable=redefined-builtin
import tensorflow as tf
from tensorflow.contrib.learn.python.learn.datasets.mnist import read_data_sets

sess = tf.InteractiveSession()

x = tf.placeholder(tf.float32, shape=[None, 784])
y_ = tf.placeholder(tf.float32, shape=[None, 10])
W = tf.Variable(tf.zeros([784,10]))
b = tf.Variable(tf.zeros([10]))
sess.run(tf.initialize_all_variables())

y = tf.nn.softmax(tf.matmul(x,W) + b)
cross_entropy = tf.reduce_mean(-tf.reduce_sum(y_ * tf.log(y), reduction_indices=[1]))
train_step = tf.train.GradientDescentOptimizer(0.5).minimize(cross_entropy)
for i in range(1000):
  batch = mnist.train.next_batch(100)
  train_step.run(feed_dict={x: batch[0], y_: batch[1]})

correct_prediction = tf.equal(tf.argmax(y,1), tf.argmax(y_,1))
accuracy = tf.reduce_mean(tf.cast(correct_prediction, tf.float32))
print(accuracy.eval(feed_dict={x: mnist.test.images, y_: mnist.test.labels}))

def weight_variable(shape):
  initial = tf.truncated_normal(shape, stddev=0.1)
  return tf.Variable(initial)

def bias_variable(shape):
  initial = tf.constant(0.1, shape=shape)
  return tf.Variable(initial)
def conv2d(x, W):
  return tf.nn.conv2d(x, W, strides=[1, 1, 1, 1], padding='SAME')

def max_pool_2x2(x):
  return tf.nn.max_pool(x, ksize=[1, 2, 2, 1],
                        strides=[1, 2, 2, 1], padding='SAME')

W_conv1 = weight_variable([5, 5, 1, 32])
b_conv1 = bias_variable([32])

x_image = tf.reshape(x, [-1,28,28,1])

h_conv1 = tf.nn.relu(conv2d(x_image, W_conv1) + b_conv1)
h_pool1 = max_pool_2x2(h_conv1)

W_conv2 = weight_variable([5, 5, 32, 64])
b_conv2 = bias_variable([64])

h_conv2 = tf.nn.relu(conv2d(h_pool1, W_conv2) + b_conv2)
h_pool2 = max_pool_2x2(h_conv2)

W_fc1 = weight_variable([7 * 7 * 64, 1024])
b_fc1 = bias_variable([1024])

h_pool2_flat = tf.reshape(h_pool2, [-1, 7*7*64])
h_fc1 = tf.nn.relu(tf.matmul(h_pool2_flat, W_fc1) + b_fc1)

keep_prob = tf.placeholder(tf.float32)
h_fc1_drop = tf.nn.dropout(h_fc1, keep_prob)

W_fc2 = weight_variable([1024, 10])
b_fc2 = bias_variable([10])

y_conv=tf.nn.softmax(tf.matmul(h_fc1_drop, W_fc2) + b_fc2)

cross_entropy = tf.reduce_mean(-tf.reduce_sum(y_ * tf.log(y_conv), reduction_indices=[1]))
train_step = tf.train.AdamOptimizer(1e-4).minimize(cross_entropy)
correct_prediction = tf.equal(tf.argmax(y_conv,1), tf.argmax(y_,1))
accuracy = tf.reduce_mean(tf.cast(correct_prediction, tf.float32))

sess.run(tf.initialize_all_variables())

for i in range(200):
  batch = mnist.train.next_batch(50)
  if i%100 == 0:
    train_accuracy = accuracy.eval(feed_dict={
        x:batch[0], y_: batch[1], keep_prob: 1.0})
    print("step %d, training accuracy %g"%(i, train_accuracy))
  train_step.run(feed_dict={x: batch[0], y_: batch[1], keep_prob: 0.5})

print("test accuracy %g"%accuracy.eval(feed_dict={x: mnist.test.images, y_: mnist.test.labels, keep_prob: 1.0}))

Please, help! Thanks!!

**Comment by girving** [comment]
I don't think we have enough information.  Is that the full error message?

**Comment by adonisDias** [comment]
Yes, jupyter notebook shows just this message on the screen. I don't know if it has a different kind of log error output.

I did a little change in the code, specific in that last line. The new line is:

teste = mnist.test.next_batch(1500)
print("test accuracy %g"%accuracy.eval(feed_dict={x:teste[0], y_: teste[1], keep_prob: 1.0}))

And the code ran fine. 
I am suspicious that is a memory problem. 
What do you think?

**Comment by girving** [comment]
I have no familiarity with Jupyter notebooks.  @colah Any guesses?

**Comment by adonisDias** [comment]
Hi Guys!!
The problem is solved. Really, it was a memory problem. 
I installed ubuntu in my notebook (8 GB RAM) and the code ran fine (total memory used during the execution).
previously i was running in a virtual machine.
Thanks!!

**Comment by girving** [comment]
Glad you figured it out!

**Comment by AndrewLouw** [comment]
This was not a memory issue - the issue arises when you install other packages , i've not worked out the details yet but I think either PIL (pillow) or pycat can cause some sort of conflict if installed in the same environment as tensorflow specifically causing this line:
from tensorflow.examples.tutorials.mnist import input_data
to cause the kernel to die. I've seen reports that cuda also causes this but it isn't a package i have installed.
My reasons for saying this is that I had a notebook working fine until i tried to add functionality - this led me to install pycat and pillow.  Now that line causes the kernel to hang until it dies even if it is the first line in the notebook - unfortunately the installs are difficult to reverse and i'm still looking for a solution.
